src/Service.Tests/Unittests/SerializationDeserializationTests.cs

```diff
@@ -247,6 +254,63 @@ public void TestColumnDefinitionNegativeCases()
             Assert.IsFalse(_columnDefinition.Equals(col));
         }
 
+        /// <summary>
+        /// Validates serialization and deserilization of Dictionary containing DatabaseTable
+        /// this is how we serialize and deserialize metadataprovider.EntityToDatabaseObject dict
+        /// </summary>
+        [TestMethod]
+        public void TestDictionaryDatabaseObjectSerializationDeserialization()
+        {
+            InitializeObjects();
+
+            Dictionary<string, DatabaseObject> dict = new() { { "person", _databaseTable } };
+
+            string serializedDict = JsonSerializer.Serialize(dict, _options);
+            Dictionary<string, DatabaseObject> deserializedDict = JsonSerializer.Deserialize<Dictionary<string, DatabaseObject>>(serializedDict, _options)!;
+
+            DatabaseTable deserializedDatabaseTable = (DatabaseTable)deserializedDict["person"];
+
+            Assert.AreEqual(deserializedDatabaseTable.SourceType, _databaseTable.SourceType);
+            Assert.AreEqual(deserializedDatabaseTable.FullName, _databaseTable.FullName);
+            deserializedDatabaseTable.Equals(_databaseTable);
+            VerifySourceDefinitionSerializationDeserialization(deserializedDatabaseTable.SourceDefinition, _databaseTable.SourceDefinition, "FirstName");
+            VerifySourceDefinitionSerializationDeserialization(deserializedDatabaseTable.TableDefinition, _databaseTable.TableDefinition, "FirstName");
+        }
+
+        /// <summary>
+        /// Test to check the impact of version change in TypeName
```

Comment:
any examples to provide of what version string looks like before and after an update? 
The description says you check the impact of version change.. what is the expected impact when a version is updated? e.g. exception or your code gracefully handles the new version?

Reply:
removed this test, the test passes in DAB code, but same version upgrade fails in workload. We have added a fallback code on our end if deserialization fail.